ui/text_displayer.js

```diff
@@ -227,11 +354,18 @@ shaka.ui.TextDisplayer = class {
 
     // Set white-space to 'pre-line' to enable showing line breaks in the text.
     captionsStyle.whiteSpace = 'pre-line';
-    captions.textContent = cue.payload;
     captionsStyle.backgroundColor = cue.backgroundColor;
     captionsStyle.color = cue.color;
     captionsStyle.direction = cue.direction;
 
+    // Add cue content
+    const content = this.buildCueElements_(cue.payload);
+    if (content) {
```

Comment:
This is another check that we don't seem to need.  The function will at worst return an empty array, which is truthy and we can still iterate through.

Reply:
Per above, and resolved.